servers/src/main/java/tachyon/worker/block/BlockMasterSync.java

```diff
@@ -220,4 +220,33 @@ private void resetMasterClient() {
         new MasterClient(NetworkAddressUtils.getMasterAddress(mTachyonConf),
             mMasterClientExecutorService, mTachyonConf);
   }
+
+  /**
+   * Thread to remove block from master
+   */
+  private class BlockRemover implements Runnable {
+    private BlockDataManager mBlockDataManager;
+    private long mUserId;
+    private long mBlockId;
+
+    public BlockRemover(BlockDataManager blockDataManager, long userId, long blockId) {
+      mBlockDataManager = blockDataManager;
+      mUserId = userId;
+      mBlockId = blockId;
+    }
+
+    @Override
+    public void run() {
+      try {
+        mBlockDataManager.removeBlock(mUserId, mBlockId);
+      } catch (IOException ioe) {
```

Comment:
I think in Java the order of catching exception is from most specific to
least specific ( the class hierarchy) NOT based on the logical happen
order. In this context, these three exception class do not have direct
hierarchy relation, it should be fine to order them as current.

During my change, I am wondering why we don't printout the exception stack
to give more detail information about the failure. Is that because of the
concerning about log size, format, etc?  Even though the current log helps
to understand, I guess most of the programmers are interested in the
failure stack.

Zhen



On Tue, Aug 11, 2015 at 7:48 AM, Shiqing Fan <notifications@github.com>
wrote:

> In servers/src/main/java/tachyon/worker/block/BlockMasterSync.java
> <https://github.com/amplab/tachyon/pull/1252#discussion_r36753452>:
>
> > +  private class BlockRemover implements Runnable {
> > +    private BlockDataManager mBlockDataManager;
> > +    private long mUserId;
> > +    private long mBlockId;
> > +
> > +    public BlockRemover(BlockDataManager blockDataManager, long userId, long blockId) {
> > +      mBlockDataManager = blockDataManager;
> > +      mUserId = userId;
> > +      mBlockId = blockId;
> > +    }
> > +
> > +    @Override
> > +    public void run() {
> > +      try {
> > +        mBlockDataManager.removeBlock(mUserId, mBlockId);
> > +      } catch (IOException ioe) {
>
> If an exception do thrown here, logically it will happen in the order:
> InvalidStateException, IOException, NotFoundException, the original catch
> structure is okay, better adjust it as described above.
>
> By the way, the IOException may be thrown delete to *block deleted fails*
> (see TieredBlockStore#removeBlockInternal), better improve the LOG.warn
> log message corresponding.
>
> Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/amplab/tachyon/pull/1252/files#r36753452>.
>